objects/src/accounts/account_id.rs

```diff
@@ -134,336 +108,511 @@ impl FromStr for AccountStorageMode {
+        account_id_from_felts(felts)
+    }
 
-        let seed = get_account_seed(
-            init_seed,
-            account_type,
-            AccountStorageMode::Public,
-            code_commitment,
-            storage_commitment,
-        )
-        .unwrap();
+    /// Creates an [`AccountId`] from the given felts where the felt at index 0 is the first felt
+    /// and the felt at index 2 is the second felt.
+    ///
+    /// # Warning
+    ///
+    /// Validity of the ID must be ensured by the caller. An invalid ID may lead to panics.
+    ///
+    /// # Panics
+    ///
+    /// If debug_assertions are enabled (e.g. in debug mode), this function panics if any of the ID
+    /// constraints are not met. See the [type documentation](AccountId) for details.
+    pub fn new_unchecked(elements: [Felt; 2]) -> Self {
+        let first_felt = elements[0];
+        let second_felt = elements[1];
+
+        // Panic on invalid felts in debug mode.
+        if cfg!(debug_assertions) {
+            validate_first_felt(first_felt)
+                .expect("AccountId::new_unchecked called with invalid first felt");
+            validate_second_felt(second_felt)
+                .expect("AccountId::new_unchecked called with invalid first felt");
+        }
 
-        Self::new(seed, code_commitment, storage_commitment).unwrap()
+        Self { first_felt, second_felt }
     }
 
     /// Constructs an [`AccountId`] for testing purposes with the given account type and storage
     /// mode.
     ///
     /// This function does the following:
-    /// - The bit representation of the account type and storage mode is prepended to the most
-    ///   significant byte of `bytes`.
-    /// - The 5th most significant bit is cleared.
-    /// - The lowest 5 bits are set to ensure the id has at least [`Self::MIN_ACCOUNT_ONES`].
-    /// - The bytes are then converted to a `u64` in big-endian format. Due to clearing the 5th most
-    ///   significant bit, the resulting `u64` will be a valid [`Felt`].
+    /// - Split the given bytes into a `first_felt = bytes[0..8]` and `second_felt = bytes[8..]`
+    ///   part to be used for the first and second felt, respectively.
+    /// - The least significant byte of the first felt is set to the version 0, and the given type
+    ///   and storage mode.
+    /// - The 32nd most significant bit in the first felt is cleared to ensure it is a valid felt.
+    ///   The 32nd is chosen as it is the lowest bit that we can clear and still ensure felt
+    ///   validity. This leaves the upper 31 bits to be set by the input `bytes` which makes it
+    ///   simpler to create test values which more often need specific values for the most
+    ///   significant end of the ID.
+    /// - In the second felt the anchor epoch is set to 0 and the lower 8 bits are cleared.
     #[cfg(any(feature = "testing", test))]
-    pub fn new_with_type_and_mode(
-        mut bytes: [u8; 8],
+    pub fn new_dummy(
```

Comment:
Could we get `rng` support? In most node tests (e.g. https://github.com/0xPolygonMiden/miden-node/pull/591) we really just want some number of unique account IDs.

Sometimes one would also like to specify the account type and storage mode I suppose, but in general just having a quick way to create a random account ID without any additional fuss would make test code easier to grok and create.